### DISCUSSION TITLE
Reload all maps (MultiMapFileTileSource)

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1113] [Q&A] [answered]
When I download a new map or when I want to change the language of a map I proceed by reinitializing everything in order to be able to view the new map or view the new language of one of the maps without having to close the app and reopen it.

So I follow these steps:
```
1.  for (int i = 1; i < map.layers().size(); i++)  // Remove all layer, except the first layer otherwise it doesn't work
        map.layers().remove(i);

2. MultiMapFileTileSource tileSources = loadMapFiles();
3. VectorTileLayer tileLayer = map.setBaseMap(tileSources);
4. map.layers().add(new BuildingLayer(map, tileLayer));
5. map.layers().add(new LabelLayer(map, tileLayer));
etc...
```

This procedure does not present any particular problems, except that:
- the new city names are superimposed on the old ones where i changed map language.

Question:
What is the best solution to be able to reinitialize the entire map?

**Comment by devemux86** [comment, marked as answer]
Try calling `Map.clearMap()` after removing / adding layers.

**Reply by moiatto** [reply]
Thank you!
